pkg/filter/filter.go

```diff
@@ -108,24 +119,40 @@ func JoinResourceTypeMatchers(ms ...ResourceTypeMatcher) ResourceTypeMatcher {
 
 // ResourceRequirement allows specifying a resource requirement by type and/or name
 type ResourceRequirement struct {
+	// Provides the Name of the resource object
 	v1.LocalObjectReference `json:",inline,omitempty"`
+	// Provides the Group, Version, and Resource values (GVR)
 	ResourceTypeRequirement `json:",inline,omitempty"`
+	// Specifies a set of label requirements to be used as filters for matches
+	metav1.LabelSelector `json:",inline,omitempty"`
 }
 
-// Matches returns true if the specified resource name/GVR matches the requirement
-func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource) bool {
-	// If the requirement does not specify a resource name - only check the
-	// ResourceTypeRequirement i.e. GVR match
-	if r.LocalObjectReference.Name == "" {
-		return r.ResourceTypeRequirement.Matches(gvr)
+// Matches returns true if the specified resource name/GVR/labels matches the requirement
+func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource, resourceLabels map[string]string) bool {
+	// If ResourceRequirement Name/GVR value of empty string matches on any specified in parameters.
+	// If ResourceRequirement Name/GVR value specified, exact match on parameter is required.
+	if !(matches(r.Name, name) && r.ResourceTypeRequirement.Matches(gvr)) {
```

Comment:
```suggestion
	if !(matches(r.Name, name) || !r.ResourceTypeRequirement.Matches(gvr)) {
```
Please add a test case too-- in this case for name being same and GVR different.

Reply:
Good catch. You mentioned this before but in my refactoring it got lost. Test and change added.